cdap-security/src/main/java/io/cdap/cdap/security/tools/AccessTokenGeneratorService.java

```diff
@@ -218,7 +214,24 @@ public static void main(String[] args)throws Exception {
     }
 
     Injector injector = createInjector(cConf);
-    AccessTokenGeneratorService tokenGenerator = injector.getInstance(AccessTokenGeneratorService.class);
-    tokenGenerator.startAndWait();
+
+    this.handler = injector.getInstance(TokenGeneratorHttpHandler.class);
+    SConfiguration sConf = injector.getInstance(SConfiguration.class);
+    CommonNettyHttpServiceFactory httpServiceFactory = injector.getInstance(CommonNettyHttpServiceFactory.class);
```

Comment:
This might introduce the `AuthenticationChannelHandler` as a pipeline modifier to the AccessTokenGeneratorService making it impossible to get a token at runtime if security is enabled in the CConf.

Reply:
Just took another look; it might be ok since you're not making authorization requests in the AccessTokenGeneratorService, but all the same you may want to avoid using the CommonNettyHttpServiceBuilder to simplify things a bit

Reply:
Thanks, fixed